commit 23eb829b62dd4f23a2c43314915b1c5489ffeba2
Author: Tomas Pizarro <tompizmor@users.noreply.github.com>
Date:   Wed Apr 4 14:23:37 2018 +0200

    [stable/moodle] Refactor image property in values.yaml (#4647)
    
    * [stable/moodle] Refactor image properties in values.yaml
    
    * Use rolling tag and refactor pullSecrets
    
    * Fix lint issue
    
    * Bump chart version
    
    * Update Chart.yaml

diff --git a/stable/moodle/Chart.yaml b/stable/moodle/Chart.yaml
index f6b4a9c8b..3707ddcb4 100644
--- a/stable/moodle/Chart.yaml
+++ b/stable/moodle/Chart.yaml
@@ -1,5 +1,5 @@
 name: moodle
-version: 0.5.3
+version: 1.0.0
 appVersion: 3.4.2
 description: Moodle is a learning platform designed to provide educators, administrators
   and learners with a single robust, secure and integrated system to create personalised
diff --git a/stable/moodle/README.md b/stable/moodle/README.md
index 65938c909..59fa82777 100644
--- a/stable/moodle/README.md
+++ b/stable/moodle/README.md
@@ -45,48 +45,51 @@ The command removes all the Kubernetes components associated with the chart and
 
 The following table lists the configurable parameters of the Moodle chart and their default values.
 
-|              Parameter              |               Description               |                   Default                   |
-|-------------------------------------|-----------------------------------------|---------------------------------------------|
-| `image`                             | Moodle image                            | `bitnami/moodle:{VERSION}`                  |
-| `imagePullPolicy`                   | Image pull policy                       | `IfNotPresent`                              |
-| `moodleUsername`                    | User of the application                 | `user`                                      |
-| `moodlePassword`                    | Application password                    | _random 10 character alphanumeric string_   |
-| `moodleEmail`                       | Admin email                             | `user@example.com`                          |
-| `smtpHost`                          | SMTP host                               | `nil`                                       |
-| `smtpPort`                          | SMTP port                               | `nil` (but moodle internal default is 25)   |
-| `smtpProtocol`                      | SMTP Protocol (options: ssl,tls, nil)   | `nil`                                       |
-| `smtpUser`                          | SMTP user                               | `nil`                                       |
-| `smtpPassword`                      | SMTP password                           | `nil`                                       |
-| `serviceType`                       | Kubernetes Service type                 | `LoadBalancer`                              |
-| `ingress.enabled`                   | If ingress should be created            | `false`                                     |
-| `ingress.annotations`               | Any ingress annotations                 | `nil`                                       |
-| `ingress.hosts`                     | List of Ingress hosts                   | `nil`                                       |
-| `ingress.tls`                       | List of certs. If defined, https is set | `nil`                                       |
-| `affinity`                          | Set affinity for the moodle pods        | `nil`                                       |
-| `resources`                         | CPU/Memory resource requests/limits     | Memory: `512Mi`, CPU: `300m`                |
-| `persistence.enabled`               | Enable persistence using PVC            | `true`                                      |
-| `persistence.storageClass`          | PVC Storage Class for Moodle volume     | `nil` (uses alpha storage class annotation) |
-| `persistence.accessMode`            | PVC Access Mode for Moodle volume       | `ReadWriteOnce`                             |
-| `persistence.size`                  | PVC Storage Request for Moodle volume   | `8Gi`                                       |
-| `persistence.existingClaim`         | If PVC exists&bounded for Moodle        | `nil` (when nil, new one is requested)      |
-| `allowEmptyPassword`                | Allow DB blank passwords                | `yes`                                       |
-| `externalDatabase.host`             | Host of the external database           | `nil`                                       |
-| `externalDatabase.port`             | Port of the external database           | `3306`                                      |
-| `externalDatabase.user`             | Existing username in the external db    | `bn_moodle`                                 |
-| `externalDatabase.password`         | Password for the above username         | `nil`                                       |
-| `externalDatabase.database`         | Name of the existing database            | `bitnami_moodle`                            |
-| `mariadb.enabled`                   | Whether to install the mariadb chart    | `true`                                      |
-| `mariadb.mariadbDatabase`           | Database name to create                 | `bitnami_moodle`                            |
-| `mariadb.mariadbUser`               | Database user to create                 | `bn_moodle`                                 |
-| `mariadb.mariadbPassword`           | Password for the database               | `nil`                                       |
-| `mariadb.mariadbRootPassword`       | MariaDB admin password                  | `nil`                                       |
-| `mariadb.persistence.enabled`       | Enable MariaDB persistence using PVC    | `true`                                      |
-| `mariadb.persistence.storageClass`  | PVC Storage Class for MariaDB volume    | `generic`                                   |
-| `mariadb.persistence.accessMode`    | PVC Access Mode for MariaDB volume      | `ReadWriteOnce`                             |
-| `mariadb.persistence.size`          | PVC Storage Request for MariaDB volume  | `8Gi`                                       |
-| `mariadb.persistence.existingClaim` | If PVC exists&bounded for MariaDB       | `nil` (when nil, new one is requested)      |
-| `mariadb.affinity`                  | Set affinity for the MariaDB pods       | `nil`                                       |
-| `mariadb.resources`                 | CPU/Memory resource requests/limits     | Memory: `256Mi`, CPU: `250m`                |
+|              Parameter              |               Description               |                   Default                               |
+|-------------------------------------|-----------------------------------------|-------------------------------------------------------- |
+| `image.registry`                    | Moodle image registry                   | `docker.io`                                             |
+| `image.repository`                  | Moodle Image name                       | `bitnami/moodle`                                        |
+| `image.tag`                         | Moodle Image tag                        | `{VERSION}`                                             |
+| `image.pullPolicy`                  | Image pull policy                       | `Always` if `imageTag` is `latest`, else `IfNotPresent` |
+| `image.pullSecrets`                 | Specify image pull secrets              | `nil`                                                   |
+| `moodleUsername`                    | User of the application                 | `user`                                                  |
+| `moodlePassword`                    | Application password                    | _random 10 character alphanumeric string_               |
+| `moodleEmail`                       | Admin email                             | `user@example.com`                                      |
+| `smtpHost`                          | SMTP host                               | `nil`                                                   |
+| `smtpPort`                          | SMTP port                               | `nil` (but moodle internal default is 25)               |
+| `smtpProtocol`                      | SMTP Protocol (options: ssl,tls, nil)   | `nil`                                                   |
+| `smtpUser`                          | SMTP user                               | `nil`                                                   |
+| `smtpPassword`                      | SMTP password                           | `nil`                                                   |
+| `serviceType`                       | Kubernetes Service type                 | `LoadBalancer`                                          |
+| `ingress.enabled`                   | If ingress should be created            | `false`                                                 |
+| `ingress.annotations`               | Any ingress annotations                 | `nil`                                                   |
+| `ingress.hosts`                     | List of Ingress hosts                   | `nil`                                                   |
+| `ingress.tls`                       | List of certs. If defined, https is set | `nil`                                                   |
+| `affinity`                          | Set affinity for the moodle pods        | `nil`                                                   |
+| `resources`                         | CPU/Memory resource requests/limits     | Memory: `512Mi`, CPU: `300m`                            |
+| `persistence.enabled`               | Enable persistence using PVC            | `true`                                                  |
+| `persistence.storageClass`          | PVC Storage Class for Moodle volume     | `nil` (uses alpha storage class annotation)             |
+| `persistence.accessMode`            | PVC Access Mode for Moodle volume       | `ReadWriteOnce`                                         |
+| `persistence.size`                  | PVC Storage Request for Moodle volume   | `8Gi`                                                   |
+| `persistence.existingClaim`         | If PVC exists&bounded for Moodle        | `nil` (when nil, new one is requested)                  |
+| `allowEmptyPassword`                | Allow DB blank passwords                | `yes`                                                   |
+| `externalDatabase.host`             | Host of the external database           | `nil`                                                   |
+| `externalDatabase.port`             | Port of the external database           | `3306`                                                  |
+| `externalDatabase.user`             | Existing username in the external db    | `bn_moodle`                                             |
+| `externalDatabase.password`         | Password for the above username         | `nil`                                                   |
+| `externalDatabase.database`         | Name of the existing database           | `bitnami_moodle`                                        |
+| `mariadb.enabled`                   | Whether to install the mariadb char     | `true`                                                  |
+| `mariadb.mariadbDatabase`           | Database name to create                 | `bitnami_moodle`                                        |
+| `mariadb.mariadbUser`               | Database user to create                 | `bn_moodle`                                             |
+| `mariadb.mariadbPassword`           | Password for the database               | `nil`                                                   |
+| `mariadb.mariadbRootPassword`       | MariaDB admin password                  | `nil`                                                   |
+| `mariadb.persistence.enabled`       | Enable MariaDB persistence using PVC    | `true`                                                  |
+| `mariadb.persistence.storageClass`  | PVC Storage Class for MariaDB volume    | `generic`                                               |
+| `mariadb.persistence.accessMode`    | PVC Access Mode for MariaDB volume      | `ReadWriteOnce`                                         |
+| `mariadb.persistence.size`          | PVC Storage Request for MariaDB volume  | `8Gi`                                                   |
+| `mariadb.persistence.existingClaim` | If PVC exists&bounded for MariaDB       | `nil` (when nil, new one is requested)                  |
+| `mariadb.affinity`                  | Set affinity for the MariaDB pods       | `nil`                                                   |
+| `mariadb.resources`                 | CPU/Memory resource requests/limits     | Memory: `256Mi`, CPU: `250m`                            |
 
 The above parameters map to the env variables defined in [bitnami/moodle](http://github.com/bitnami/bitnami-docker-moodle). For more information please refer to the [bitnami/moodle](http://github.com/bitnami/bitnami-docker-moodle) image documentation.
 
diff --git a/stable/moodle/templates/deployment.yaml b/stable/moodle/templates/deployment.yaml
index 085839692..565b1a448 100644
--- a/stable/moodle/templates/deployment.yaml
+++ b/stable/moodle/templates/deployment.yaml
@@ -16,11 +16,17 @@ spec:
       {{- if .Values.affinity }}
       affinity:
 {{ toYaml .Values.affinity | indent 8 }}
+      {{- end }}
+      {{- if .Values.image.pullSecrets }}
+      imagePullSecrets:
+      {{- range .Values.image.pullSecrets }}
+        - name: {{ . }}
+      {{- end}}
       {{- end }}
       containers:
       - name: {{ template "moodle.fullname" . }}
-        image: "{{ .Values.image }}"
-        imagePullPolicy: {{ default "" .Values.imagePullPolicy | quote }}
+        image: "{{ .Values.image.registry }}/{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
         env:
         - name: ALLOW_EMPTY_PASSWORD
           value: {{ .Values.allowEmptyPassword | quote }}
diff --git a/stable/moodle/values.yaml b/stable/moodle/values.yaml
index cbce4697e..35d15cd85 100644
--- a/stable/moodle/values.yaml
+++ b/stable/moodle/values.yaml
@@ -1,13 +1,21 @@
-## Bitnami Moodle` image version
+## Bitnami Moodle image version
 ## ref: https://hub.docker.com/r/bitnami/moodle/tags/
 ##
-image: bitnami/moodle:3.4.2-r0
-
-## Specify a imagePullPolicy
-## Defaults to 'Always' if image tag is 'latest', else set to 'IfNotPresent'
-## ref: http://kubernetes.io/docs/user-guide/images/#pre-pulling-images
-##
-imagePullPolicy: IfNotPresent
+image:
+  registry: docker.io
+  repository: bitnami/moodle
+  tag: 3.4.2
+  ## Specify a imagePullPolicy
+  ## Defaults to 'Always' if image tag is 'latest', else set to 'IfNotPresent'
+  ## ref: http://kubernetes.io/docs/user-guide/images/#pre-pulling-images
+  ##
+  pullPolicy: IfNotPresent
+  ## Optionally specify an array of imagePullSecrets.
+  ## Secrets must be manually created in the namespace.
+  ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/
+  ##
+  # pullSecrets:
+  #   - myRegistrKeySecretName
 
 ## User of the application
 ## ref: https://github.com/bitnami/bitnami-docker-moodle#configuration
